src/Aspire.Hosting/ApplicationModel/ExistingResourceAnnotation.cs

```diff
@@ -0,0 +1,23 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Represents a resource that is not managed by Aspire's provisioning or
+/// container management layer.
+/// </summary>
+public class ExistingResourceAnnotation(string name, bool isPublishMode = false) : IResourceAnnotation
+{
+    /// <summary>
+    /// The name of the existing resource.
+    /// </summary>
+    public string Name { get; } = name;
+
+    /// <summary>
+    /// True if the existing resource is reference in publish mode.
+    /// </summary>
+    public bool IsPublishMode { get; } = isPublishMode;
```

Comment:
Why is this needed?

Reply:
To distinguish between `RunAsExisting` vs `PublishAsExisting`.

Reply:
I don't this that needs to be part of the annotation. This should just have the data required to reference the resource. It doesn't need the mode. Run/PublishAs are just idioms in the app model driven by the current ExecutionContext mode.

Reply:
How would we indicate that behavior needs to be gated on the `ExecutionContext` in this case?

Reply:
I see why you say that becase these are implemented as a generic annotation instead of methods on the specific builder. Maybe we should reconsider that?

Reply:
Actually, I'm dumb this is much simpler than that. Is RunAsExisting, check the mode, if it's run add the attribute, same for PublishAsExisting. The annotation doesn't need the mode, it just needs to know if there's data about where this resource is.